src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2362,37 +2362,73 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
         public override BoundNode VisitConditionalOperator(BoundConditionalOperator node)
         {
-            var isByRef = node.IsRef;
-
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
+            bool consequenceReachable = this.StateWhenTrue.Reachable;
+            bool alternativeReachable = this.StateWhenFalse.Reachable;
+
+            TypeWithState consequenceRValue;
+            TypeWithState alternativeRValue;
+
+            if (node.IsRef)
+            {
+                TypeWithAnnotations consequenceLValue;
+                TypeWithAnnotations alternativeLValue;
+                (consequenceLValue, consequenceRValue) = visitConditionalRefOperand(consequenceState, node.Consequence);
+                consequenceState = this.State;
```

Comment:
>this [](start = 35, length = 4)

You should take the reachability here, after visiting the operand.  That is needed e.g. to handle an operand that can't complete normally.

Reply:
`b ? M1(false ? e : throw) : M2(e)`

---
In reply to: [268371523](https://github.com/dotnet/roslyn/pull/34323#discussion_r268371523) [](ancestors = 268371523)